CONTRIBUTING.md

```diff
@@ -1,43 +1,40 @@
+If you plan to submit a pull request, we ask that you first create an [issue](https://github.com/buildkite-plugins/monorepo-diff-buildkite-plugin/issues). For new features or modifications to existing functionality, please start a discussion with the maintainers. For straightforward bug fixes, an issue is enough without a preliminary discussion.
 
 ## Developing
 
-- [Install Go.](https://golang.org/doc/install)
-- The project uses Makefile. Install `make` command.
-- Fork this repository.
-- Clone the forked repository.
--  Make changes (see [Formatting](https://github.com/buildkite-plugins/monorepo-diff-buildkite-plugin/blob/master/CONTRIBUTING.md#formatting)) and commit to your fork. Commit messages should follow the [Conventional Commits](https://www.conventionalcommits.org/) style.
-- Add appropriate unit tests (see [Testing](https://github.com/buildkite-plugins/monorepo-diff-buildkite-plugin/blob/master/CONTRIBUTING.md#testing)) for your changes.
-- Update documentation if appropriate.
-- Create a pull request with your changes.
-- Github action will run the necessary checks against your pull request.
-- A maintainer will review the pull request once all checks are
-- A maintainer will merge and create a release (see [Releasing](https://github.com/buildkite-plugins/monorepo-diff-buildkite-plugin/blob/master/CONTRIBUTING.md#releasing)).
+To get started with development:
+
+- [Install Go](https://golang.org/doc/install).
+- Ensure you have `make` installed, as this project uses a Makefile.
+- Fork this repository and clone your fork locally.
+- Make your changes (see [Formatting](https://github.com/buildkite-plugins/monorepo-diff-buildkite-plugin/blob/master/CONTRIBUTING.md#formatting)) and commit to your fork. Use [Conventional Commits](https://www.conventionalcommits.org/) style for commit messages.
+- Add relevant unit tests (see [Testing](https://github.com/buildkite-plugins/monorepo-diff-buildkite-plugin/blob/master/CONTRIBUTING.md#testing)) for your changes.
+- Update documentation if necessary.
+- Open a pull request; required checks will automatically run on your changes.
+- A maintainer will review your pull request once all checks pass.
+- After review and approval, a maintainer will merge your pull request and create a release (see [Releasing](https://github.com/buildkite-plugins/monorepo-diff-buildkite-plugin/blob/master/CONTRIBUTING.md#releasing)).
 
 ## Testing
 
-All changes must be unit tested and meet the project test coverage threshold (73%) requirement.
-Run `make test` to run all tests and generate coverage reports before submitting a pull request.
+All changes must be unit tested, and test coverage should meet the project's minimum threshold of 73%. Run `make test` to execute all tests and generate coverage reports before submitting a pull request.
```

Comment:
Do we do anything with these coverage reports? It may be good to have those added to the build as an artifact/annotation so that we can also check them and make sure we have that coverage.

Reply:
For now we don't export it anywhere just to check if its up to the threshold. Will look into either options